lib/parser/source/tree_rewriter.rb

```diff
@@ -355,21 +365,32 @@ def insert_after_multi(range, text)
 
       extend Deprecation
 
+      ##
+      # @api private
+      # reserved for TreeAction
+      #
+      def enforce_policy(event)
+        return if policy(event) == :accept
+        return unless (values = yield)
+        trigger_policy(event, **values)
+      end
+
       protected
 
       attr_reader :action_root
 
       private
 
-      ACTIONS = %i[accept warn raise].freeze
-      def check_policy_validity
-        invalid = @policy.values - ACTIONS
-        raise ArgumentError, "Invalid policy: #{invalid.join(', ')}" unless invalid.empty?
+      ACTIONS = %i[accept warn raise].to_set.freeze
```

Comment:
Is it actually better to use sets for 3 symbols? IIRC sets in ruby are hashes, and small hashes are arrays. Do you get anything from this `to_set`? I thought it makes it slower because computing `.hash` and comparing it is slower than comparing symbols that are numbers internally. Not a blocker at all

Reply:
Yeah, good point. I'm not sure I can explain why, but even for a three element set, a non matching include check is faster on a `Set` than on an `Array`, so is a matching check on the 2nd or 3rd element. Only a matching check on the first element seems faster on an `Array`.
Beyond performance, I feel that an object to call `include?` on should be a `Set`, and `Set` should optimize that call...